docs/source/configuration/authorization.mdx

```diff
@@ -98,199 +140,263 @@ So the `me` top level operation would not even be executed. When returning the r
+You can define the `@requiredScopes` directive on any subgraph schema like this:
 
-Depending on which scope set is presented by the request, different parts of the query may be available.
+```graphql
+directive @requiresScopes(scopes: [String!]!) on OBJECT | FIELD_DEFINITION | INTERFACE | SCALAR | ENUM
+```
+
+The directive should include a `scopes` argument that expects an array of required scopes.
+
+The directive validates the required scopes by loading the object at the `apollo_authentication::JWT::claims` key in a request's context.
+That object's `scope` key should contain a space separated list of scopes in the format defined by [OAuth2 access token scopes](https://datatracker.ietf.org/doc/html/rfc6749#section-3.3).
+
+Depending on the scopes present on the request, the router filters out unauthorized fields and types.
+If a field's `scopes` array includes multiple scopes, the request must include _all_ required scopes to resolve it.
+
+#### Example `@requiresScopes` use case
+
+Imagine your social media platform has an ecommerce component.
+You need to be able to query for the following:
 
-Assuming we have this schema:
+- An authenticated user's profile (`me`)
+- Other user's emails
+- Products and the users who have ordered it
+
+Your schema may look something like this:
 
 ```graphql
 type Query {
-    me: User @requiresScopes(scopes: ["profile:read"])
-    products: [Products]
+  me: User @requiresScopes(scopes: ["profile:read"])
+  products: [Products]
 }
 
 type User {
-    id: ID
-    name: String
-    email: @requiresScopes(["user:read"])
+  id: ID
+  name: String
+  email: @requiresScopes(["user:read"])
 }
 
 type Product {
-    id: ID
-    name: String
-    amount: Int @requiresScopes(["employee", "inventory"])
-    orders: [User] @requiresScopes(["employee"])
+  id: ID
+  name: String
+  amount: Int @requiresScopes(["inventory:read"])
+  orders: [User] @requiresScopes(["user:read", "inventory:read"])
 }
 ```
 
-And this query:
+The router executes the following query differently, depending on the request's attached scopes:
 
 ```graphql
-{
-    me {
-        name
-        email
-    }
+query {
+  me {
+    name
+    email
+  }
 
-    products {
-        name
-        amount
-        orders {
-            name
-        }
+  products {
+    name
+    amount
+    orders {
+      name
     }
+  }
 }
 ```
 
-If the request presented the scopes `profile:read user:read`, then it would be interpreted as:
+If the request includes the `profile:read user:read` scope set, then the router would execute the following filtered query: 
 
+```graphql title="Request scopes: 'profile:read user:read'"
+query {
+  me {
+    name
+    email
+  }
 
-```graphql
-{
-    me {
-        name
-        email
-    }
+  products {
+    name
+    /* The following fields are filtered out:
+    amount # Requires "inventory:read" scope
+    orders # Requires "user:read" and "inventory:read" scopes
+    */
```

Comment:
I might split this and say "this is what was split out" to make it clearer what was and wasn't executed.

Reply:
Good call, I think these code comparisons could be clearer.

Reply:
https://github.com/apollographql/router/pull/3449/commits/e006b24cdeb767339f0e66d44cebf0ab8769c463